lib/ash_phoenix/live_view.ex

```diff
@@ -603,4 +645,21 @@ defmodule AshPhoenix.LiveView do
   defp assign(socket, one, two) do
     Phoenix.Socket.assign(socket, one, two)
   end
+
+  @doc """
+  Shorthand to add results of a page onto a socket along with the page.
+
+  ### Examples:
+
+  ```elixir
+  AshPhoenix.LiveView.assign_page_and_stream_result(%Phoenix.LiveView.Socket{}, %Ash.Page.Offset{results: [1,2,3]})
+  # => %Phoenix.LiveView.Socket{assigns: %{results: [1,2,3], page: %Ash.Page.Offset{results: nil}}}
+  ```
+  """
+  @spec assign_page_and_stream_result(Phoenix.LiveView.Socket.t(), Ash.Page.page(), atom()) ::
+          Phoenix.LiveView.Socket.t()
+  def assign_page_and_stream_result(socket, page, result_key \\ :results) do
+    {results, page} = Map.get_and_update(page, :results, &{&1, nil})
+    Phoenix.Component.assign(socket, [{result_key, results}, {:page, page}])
```

Comment:
By having any static key here that means you can only use this utility once-per-socket right? Feels like both keys should be required to me.

Reply:
Oh. You want to make the page key customizable? Can do